src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -880,6 +887,29 @@ export class HeaderFilter<T extends Slick.SlickData> {
             }
         }
     }
+
+    private setSortButtonImage(
+        $sortButton: JQuery<HTMLElement>,
+        column: FilterableColumn<T>,
+    ) {
+        if (
+            $sortButton &&
+            column.sorted &&
+            column.sorted !== SortProperties.NONE
+        ) {
+            switch (column.sorted) {
+                case SortProperties.ASC:
+                    $sortButton.removeClass("slick-header-sort-button");
+                    $sortButton.addClass("slick-header-sortasc-button");
+                    break;
+                case SortProperties.DESC:
+                    $sortButton.removeClass("slick-header-sort-button");
+                    $sortButton.addClass("slick-header-sortdesc-button");
+                    break;
+            }
+            this.columnSortButtonMapping.set(column.id!, column.sorted);
```

Comment:
It's a bit confusing to have the state mgmt call within `setSortButtonImage`, which by the name sounds like a DOM operation function.
Where do we do the similar thing for filter? I don't see a similar state seeting in the setFilterButtonImage func

Reply:
I can move it out, we don't do something like that for filter since it's just either a filled icon or not filled